### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by juandbc**
## Summary
- Add spring security dependency
- Add password attribute to User model
- Implemented custom security configuration to retrieve user credentials from the database
- Refactor test to avoid duplication
- Add tests for controllers

## Trello Card
https://trello.com/c/CzY85bvD/23-integrar-spring-security

### CODE REVIEW

File: src/main/java/wolox/training/models/User.java [hunk 1]
```diff
@@ -64,6 +65,10 @@
     @Getter
     private String username;
 
+    @Getter
+    @Setter
+    private String password;
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/models/User.java:73]: add `@JsonProperty(access = Access.WRITE_ONLY)`, to not show the password when the user is consulted

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/models/User.java (hunk 1 above)
Comment (juandbc) [reply src/main/java/wolox/training/models/User.java:73]: I'll apply it, thanks.

Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/User.java:73]: Add a PUT endpoint into UserController that allows updating only the password.

The password must always be modified using a resource other than updating a User

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Very good implementation, I just leave a comment 😄

**Review by jeymarsonwolox — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**